Applications/Spire/Include/Spire/OrderImbalanceIndicator/CachedOrderImbalanceIndicatorModel.hpp

```diff
@@ -0,0 +1,61 @@
+#ifndef SPIRE_CACHED_ORDER_IMBALANCE_INDICATOR_MODEL_HPP
+#define SPIRE_CACHED_ORDER_IMBALANCE_INDICATOR_MODEL_HPP
+#include <unordered_set>
+#include <boost/icl/continuous_interval.hpp>
+#include <boost/icl/interval_set.hpp>
+#include "Spire/OrderImbalanceIndicator/OrderImbalanceIndicatorModel.hpp"
+
+namespace Spire {
+
+  //! Represents an OrderImbalanceModel that caches loaded order imbalances
+  //! from a source model.
+  class CachedOrderImbalanceIndicatorModel :
+      public OrderImbalanceIndicatorModel {
+    public:
+
+      //! Constructs a CachedOrderImbalanceModel with a given source model.
+      /*
+        \param source_model The model supplying data to the cached model.
+      */
+      CachedOrderImbalanceIndicatorModel(
+        std::shared_ptr<OrderImbalanceIndicatorModel> source_model);
+
+      OrderImbalanceIndicatorModel::SubscriptionResult subscribe(
```

Comment:
You can remove the ```OrderImbalanceIndicatorModel::```, anything in the ```OrderImbalanceIndicatorModel``` namespace/scope will implicitly be available to the ```CachedOrderImbalanceIndicatorModel```.

Reply:
I get an error about a missing type specifier when I leave it out.

Reply:
Determine the cause of the compiler error and resolve it. Just like you don't write out ```OrderImbalanceIndicatorModel::OrderImbalanceSignal::slot_type``` you also don't write out ```OrderImbalanceIndicatorModel::SubscriptionResult```.

Reply:
Do you mean this for just the header file?